internal/transport/http2_server.go

```diff
@@ -266,6 +266,13 @@ func newHTTP2Server(conn net.Conn, config *ServerConfig) (_ ServerTransport, err
 	// Check the validity of client preface.
 	preface := make([]byte, len(clientPreface))
 	if _, err := io.ReadFull(t.conn, preface); err != nil {
+		// In deployments where a gRPC server runs behind a cloud load balancer
+		// which performs regular TCP level health checks, the connection is
+		// closed immediately by the latter. Skipping the error here will help
+		// reduce log clutter.
+		if err == io.EOF {
```

Comment:
What about `io.ErrUnexpectedEOF`?

Reply:
Is this a regression?  Maybe the logs were happening before that refactor, too?  I would think either error could be safely skipped.

Reply:
That seems reasonable.